t/t1092-sparse-checkout-compatibility.sh

```diff
@@ -588,6 +588,144 @@ test_expect_success 'reset with wildcard pathspec' '
 	test_all_match git status --porcelain=v2
 '
 
+# NEEDSWORK: although update-index executes without error on files outside
+# the sparse checkout definition, it does not actually add the file to the
+# index. This is also true when "--no-ignore-skip-worktree-entries" is
+# specified.
+test_expect_success 'update-index add outside sparse definition' '
+	init_repos &&
+
+	write_script edit-contents <<-\EOF &&
+	echo text >>$1
+	EOF
+
+	run_on_sparse mkdir -p folder1 &&
+	run_on_sparse cp ../initial-repo/folder1/a folder1/a &&
+
+	# Edit the file only in sparse checkouts so that, when checking the status
+	# of the index, the unmodified full-checkout is compared to the "modified"
+	# sparse checkouts.
+	run_on_sparse ../edit-contents folder1/a &&
+
+	test_sparse_match git update-index --add folder1/a &&
+	test_all_match git status --porcelain=v2 &&
+	test_sparse_match git update-index --add --no-ignore-skip-worktree-entries folder1/a &&
+	test_all_match git status --porcelain=v2
+'
+
+test_expect_success 'update-index remove outside sparse definition' '
+	init_repos &&
+
+	# When --remove is specified, files outside the sparse checkout definition
+	# are considered "removed".
+	rm -f full-checkout/folder1/a &&
+	test_all_match git update-index --remove folder1/a &&
+	test_all_match git status --porcelain=v2 &&
+
+	git reset --hard &&
+
+	# When --ignore-skip-worktree-entries is explicitly specified, a file
+	# outside the sparse definition is not added to the index as "removed"
+	# (thus matching the unmodified full-checkout).
+	test_sparse_match git update-index --remove --ignore-skip-worktree-entries folder1/a &&
+	test_all_match git status --porcelain=v2 &&
+
+	git reset --hard &&
+
+	# --force-remove supercedes --ignore-skip-worktree-entries and always
+	# removes the file from the index.
+	test_all_match git update-index --force-remove --ignore-skip-worktree-entries folder1/a &&
+	test_all_match git status --porcelain=v2
+'
+
+test_expect_success 'update-index folder add/remove' '
+	init_repos &&
+
+	test_all_match test_must_fail git update-index --add --remove deep &&
```

Comment:
I'm a little confused on this usage of `--add` and `--remove`. Would you mind helping me understand the purpose of using both in the same command?

Reply:
Yep! According to [the doc for `update-index`](https://github.com/microsoft/git/blob/vfs-2.33.0/Documentation/git-update-index.txt#L44-L52), the `--add` and `--remove` flags indicate that, if a file is brand-new or removed from the filesystem (respectively), those files should not be skipped. 

For example, if you remove `file1` in your local working copy, then run `git update-index file1`, it won't be added to the index as "deleted" (staged for removal). However, if you run `git update-index --remove file1`, it _will_ be removed.

I added both `--add` and `--remove` to this `update-index` execution to show that, even with _no_ file change types skipped (even complete addition or removal), this command still fails.

Reply:
Awesome, thank you!